plans/apply.pp

```diff
@@ -0,0 +1,31 @@
+plan terraform::apply(
+	Optional[String[1]] $dir = undef,
+	Optional[String[1]] $state = undef,
+	Optional[Variant[String[1], Array[String[1]]]] $target = undef,
+	Optional[Hash] $var = undef,
+	Optional[String[1]] $var_file = undef,
+	Optional[Boolean] $return_output = false
+  ) {
+
+  $apply_opts = {
+  	'dir' => $dir,
+  	'state' => $state,
```

Comment:
The `resolve_references` task uses the `statefile` key for the path to load state from -- since these are all part of the same module we should probably be consistent and use the same key across all of the tasks/plans in the module.

Reply:
I wanted to map directly to the actual cli option names. Changing it in the other task is a breaking change. Not sure what the right thing to do about it is.

Reply:
Inventory v2 is still considered experimental in our docs, so would that apply to plugins as well? It'd be nice to have consistent keys... If we don't want to have breaking changes, would a workaround be accepting either `state` or `statefile` in `resolve_reference` but only documenting `state` work?

Reply:
I think that the question is what do we care more about, that task options are consistent or that task options map to terraform options. I think both are important. So my vote would be to change state to statefile in resolve_reference task.

Reply:
My vote would also go towards updating `resolve_reference` to take `state`.

Reply:
We will update resolve_reverence task to use `state`. this is part of https://github.com/puppetlabs/bolt/issues/1405 and will be rolled out with that breaking change.